libbeat/common/docker/watcher.go

```diff
@@ -316,7 +338,10 @@ func (w *watcher) listContainers(options types.ContainerListOptions) ([]*Contain
 		// If there are no network interfaces, assume that the container is on host network
 		// Inspect the container directly and use the hostname as the IP address in order
```

Comment:
This assumption is actually incorrect. Docker can have some of known network modes: default (bridge), host and container (usually used in cloud platforms like kubernetes).

It can be determined by checking .HostConfig.NetworkMode property:

```
$ sudo curl -s --fail -m3 -XGET --unix-socket /var/run/docker.sock http://localhost/containers/json | jq -r .[].HostConfig.NetworkMode
container:b79efb8d2710b3fb2ae6b9fff5e1cf6f8e841301df9a355371c197156d72c3d4
container:a7fb40e3f13684e91f26c1ed26096e0d39df5d7860db31bab5f924cec5838366
default
default
default
host
container:9f187b5c30cb2ceb43907a2513e093f1039be42c230c33d70a601e9c3173dc5d
default
default
```

if network mode is `container:<ID>` this `<ID>` should be inspected to fetch from standard config `.NetworkSettings.Networks` like above.

But this is subject for separate PR.

Reply:
This sounds about right! are you willing to open a new PR? :innocent:

Reply:
I will try to take a shot on this next week